### ISSUE TITLE
Error thrown when doing client side redirect

### ISSUE THREAD

**Issue by dehamzah** [gatsbyjs/gatsby#23955]
## Description

Error thrown in development when doing client side redirection using `Redirect` component from `@reach/router`.

```
TypeError: Cannot read property 'innerText' of null
(anonymous function)
/sandbox/.cache/navigation.js:177
  174 |   pageName = pageHeadings[0].textContent
  175 | }
  176 | const newAnnouncement = `Navigated to ${pageName}`
> 177 | const oldAnnouncement = this.announcementRef.current.innerText
  178 | if (oldAnnouncement !== newAnnouncement) {
  179 |   this.announcementRef.current.innerText = newAnnouncement
  180 | }
View compiled
```

### Steps to reproduce

I already create code example here: https://codesandbox.io/s/kind-jones-uter0?file=/src/pages/c.js

Click link "Go to page /c" and wait a few seconds. It will thrown an error like i mention above.

### Expected result

It should redirect successfully without throwing an error

### Actual result

Error thrown. `TypeError: Cannot read property 'innerText' of null`

### Environment

```
  System:
    OS: macOS 10.15.4
    CPU: (4) x64 Intel(R) Core(TM) i7-5650U CPU @ 2.20GHz
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 10.16.3 - ~/.nvm/versions/node/v10.16.3/bin/node
    Yarn: 1.21.0 - ~/.nvm/versions/node/v10.16.3/bin/yarn
    npm: 6.14.5 - ~/.nvm/versions/node/v10.16.3/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 81.0.4044.138
    Safari: 13.1
  npmGlobalPackages:
    gatsby-cli: 2.11.12
```
